<a href="https://colab.research.google.com/github/anunyathpn/PythonDataViz2026/blob/main/HW7.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

HW

##สร้างตารางใหม่ ที่ค่าใน ```sex``` เป็น missing ทั้งหมด  
* สรุปว่าทำไม record นั้นๆถึงเป็น missing

In [1]:
import pandas as pd
import numpy as np


In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [5]:
import os
path = '/content/drive/My Drive/PythonDataViz2026'
file_path = os.path.join(path,'confirmed-cases.csv')
print(file_path)

/content/drive/My Drive/PythonDataViz2026/confirmed-cases.csv


In [8]:
df = pd.read_csv(file_path)
df.head()

/tmp/ipykernel_1127/2641854224.py:1: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(file_path)


,No.,announce_date,Notified date,sex,age,Unit,nationality,province_of_isolation,risk,province_of_onset,district_of_onset
0,1,12/1/2020,NaN,หญิง,61.0,ปี,China,กรุงเทพมหานคร,คนต่างชาติเดินทางมาจากต่างประเทศ,กรุงเทพมหานคร,NaN
1,2,17/1/2020,NaN,หญิง,74.0,ปี,China,กรุงเทพมหานคร,คนต่างชาติเดินทางมาจากต่างประเทศ,กรุงเทพมหานคร,NaN
2,3,22/1/2020,NaN,หญิง,73.0,ปี,Thailand,นครปฐม,คนต่างชาติเดินทางมาจากต่างประเทศ,นครปฐม,เมือง
3,4,22/1/2020,NaN,ชาย,68.0,ปี,China,กรุงเทพมหานคร,คนต่างชาติเดินทางมาจากต่างประเทศ,กรุงเทพมหานคร,NaN
4,5,24/1/2020,NaN,หญิง,66.0,ปี,China,นนทบุรี,คนต่างชาติเดินทางมาจากต่างประเทศ,กรุงเทพมหานคร,NaN


In [9]:
df_missing_sex = df[df['sex'].isna()]
df_missing_sex

,No.,announce_date,Notified date,sex,age,Unit,nationality,province_of_isolation,risk,province_of_onset,district_of_onset
4397,4398,20/12/2020,NaN,NaN,NaN,NaN,Burma,สมุทรสาคร,Cluster สมุทรสาคร,สมุทรสาคร,เมือง
4398,4399,20/12/2020,NaN,NaN,NaN,NaN,Burma,สมุทรสาคร,Cluster สมุทรสาคร,สมุทรสาคร,เมือง
4399,4400,20/12/2020,NaN,NaN,NaN,NaN,Burma,สมุทรสาคร,Cluster สมุทรสาคร,สมุทรสาคร,เมือง
4400,4401,20/12/2020,NaN,NaN,NaN,NaN,Burma,สมุทรสาคร,Cluster สมุทรสาคร,สมุทรสาคร,เมือง
4401,4402,20/12/2020,NaN,NaN,NaN,NaN,Burma,สมุทรสาคร,Cluster สมุทรสาคร,สมุทรสาคร,เมือง
...,...,...,...,...,...,...,...,...,...,...,...
838332,838333,12/8/2021,11/8/2021,NaN,3.0,ปี,Thailand,กรุงเทพมหานคร,สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้,กรุงเทพมหานคร,บางคอแหลม
839174,839175,12/8/2021,11/8/2021,NaN,47.0,ปี,Thailand,กรุงเทพมหานคร,ระบุไม่ได้,กรุงเทพมหานคร,บางคอแหลม
839304,839305,12/8/2021,11/8/2021,NaN,NaN,NaN,NaN,กรุงเทพมหานคร,ระบุไม่ได้,NaN,NaN
839396,839397,12/8/2021,11/8/2021,NaN,NaN,NaN,NaN,นครราชสีมา,ทัณฑสถาน/เรือนจำ,นครราชสีมา,NaN


In [10]:
df_missing_sex.shape[0]

54895

**สรุป**
   มีผู้ป่วยที่ไม่ทราบเพศ 54895 ราย
   ส่วนสาเหตุที่เป็นไปได้ของการเกิด missing ที่ recordนั้นๆ คือ ความไม่ครบถ้วนของการจัดเก็บข้อมูล ความคลาดเคลื่อนขณะบันทึก และการตกหล่นในขั้นตอนการกรอกข้อมูล
